### ISSUE TITLE
Prompting simulator

### ISSUE THREAD

**Issue by bhack** [facebookresearch/segment-anything-2#114]
> We train by simulating an interactive setting, sampling 8-frame sequences and randomly selecting up to 2
frames (including the first) for corrective clicks. During training, we use ground-truth masklets and model
predictions to sample prompts, with initial prompts being the ground-truth mask (50% probability), a positive
click from the ground-truth mask (25%), or a bounding box input (25%).
We restrict the maximum number of masklets for each sequence of 8 frames to 3 randomly chosen ones. We
reverse the temporal order with a probability of 50% to help generalization to bi-directional propagation.
When we sample corrective clicks - with a small probability of 10%, we randomly sample clicks from the
ground truth mask, irrespective of the model prediction, to allow additional flexibility in mask refinement.

@ronghanghu Are you going to commit the promoting simulator?

**Comment by haithamkhedr** [comment]
Hi @bhack, we released the training code for SAM 2 (which includes how we simulate user prompts during training).
